File: app/src/main/java/se/hactar/movieregister/debug/DebugLifcycleObserver.java
package se.hactar.movieregister.debug;

import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.LifecycleObserver;
import android.arch.lifecycle.LifecycleOwner;
import android.arch.lifecycle.OnLifecycleEvent;

import timber.log.Timber;

public class DebugLifcycleObserver implements LifecycleObserver {

    private final String tag;

    public DebugLifcycleObserver(final LifecycleOwner lifecycleOwner) {
        this.tag = "<" + lifecycleOwner.getClass().getSimpleName() + "> ";
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    public void onResume() {
        Timber.d(tag + Lifecycle.Event.ON_RESUME);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    public void onPause() {
        Timber.d(tag + Lifecycle.Event.ON_PAUSE);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    public void onStart() {
        Timber.d(tag + Lifecycle.Event.ON_START);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    public void onStop() {
        Timber.d(tag + Lifecycle.Event.ON_STOP);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    public void onCreate() {
        Timber.d(tag + Lifecycle.Event.ON_CREATE);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    public void onDestroy() {
        Timber.d(tag + Lifecycle.Event.ON_DESTROY);
    }
}

File: app/src/main/java/se/hactar/movieregister/helper/DebugLifcycleObserver.kt
package se.hactar.movieregister.helper

import android.arch.lifecycle.Lifecycle
import android.arch.lifecycle.LifecycleObserver
import android.arch.lifecycle.LifecycleOwner
import android.arch.lifecycle.OnLifecycleEvent

import timber.log.Timber

class DebugLifcycleObserver(lifecycleOwner: LifecycleOwner) : LifecycleObserver {

    private val tag: String = "<" + lifecycleOwner.javaClass.simpleName + "> "

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onResume() = Timber.d(tag + Lifecycle.Event.ON_RESUME)

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() = Timber.d(tag + Lifecycle.Event.ON_PAUSE)

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart() = Timber.d(tag + Lifecycle.Event.ON_START)

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onStop() = Timber.d(tag + Lifecycle.Event.ON_STOP)

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate() = Timber.d(tag + Lifecycle.Event.ON_CREATE)

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy() = Timber.d(tag + Lifecycle.Event.ON_DESTROY)
}
